sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/CBDE/CbdeHandler.cs

```diff
@@ -366,27 +371,37 @@ private void RaiseIssuesFromJSon(CompilationAnalysisContext context)
                 jsonFileContent = File.ReadAllText(cbdeJsonOutputPath);
                 jsonIssues = JsonConvert.DeserializeObject<List<List<JObject>>>(jsonFileContent);
             }
-            catch
+            catch(Exception exception)
             {
-                LogIfFailure($"- error parsing json file {cbdeJsonOutputPath}");
-                Log(logStringBuilder.ToString());
-                Console.Error.WriteLine($"Error parsing output from CBDE, more details in {cbdeProcessSpecificPath}");
-                return;
+                if (exception is JsonException || exception is IOException)
+                {
+                    LogIfFailure($"- error parsing json file {cbdeJsonOutputPath}: {exception.ToString()}");
+                    Log(logStringBuilder.ToString());
+                    Console.Error.WriteLine($"Error parsing output from CBDE, more details in {cbdeProcessSpecificPath}");
+                    return;
+                }
+                throw exception;
             }
 
+            // in cbde json output there is enclosing {}, so this is considered as a list of list
+            // with one element in the outer list
             foreach (var issue in jsonIssues.First())
             {
                 LogIfFailure($"  * processing token {issue.ToString()}");
                 try
                 {
                     RaiseIssueFromJToken(issue, context);
                 }
-                catch
+                catch(Exception e)
                 {
-                    LogIfFailure($"  * error reporting token {cbdeJsonOutputPath}");
-                    Log(logStringBuilder.ToString());
-                    Console.Error.WriteLine($"Error raising issue from CBDE, more details in {cbdeProcessSpecificPath}");
-                    continue;
+                    if (e is SystemException || e is JsonException)
+                    {
+                        LogIfFailure($"  * error reporting token {cbdeJsonOutputPath}: {e.ToString()}");
+                        Log(logStringBuilder.ToString());
+                        Console.Error.WriteLine($"Error raising issue from CBDE, more details in {cbdeProcessSpecificPath}");
```

Comment:
this may not show in the users' console (Roslyn eats standard output)

Reply:
removed and replaced by throwing an exception